src/providers/taskwarrior.rs

```diff
@@ -0,0 +1,58 @@
+use serde::Deserialize;
+use super::Provider;
+use futures::future::BoxFuture;
+use futures::FutureExt;
+use crate::models::Todo;
+use druid::im::Vector;
+use task_hookrs::tw::query;
+
+#[derive(Debug, Clone, Deserialize)]
+pub struct TaskwarriorConfig {
+    name: Option<String>,
+    query: String
+}
+
+pub struct TaskwarriorProvider {
+    name: Option<String>,
+    query: String
+}
+
+impl TaskwarriorProvider {
+    pub fn new(config: TaskwarriorConfig) -> anyhow::Result<Self> {
+        Ok(Self {
+            name: config.name,
+            query: config.query
+        })
+    }
+
+    async fn fetch_todos(&self) -> anyhow::Result<Vector<Todo>> {
+        tracing::info!("Fetching TaskWarrior Tasks...");
+        let mut todos: Vector<Todo> = Vector::new();
+        if let Ok(tasks) = query(&self.query) {
+            for task in tasks {
+                todos.push_back(Todo{
+                    title: String::from(task.description()),
```

Comment:
Hier geht auch `task.description().into()`. Is zwar weniger explizit aber liest sich besser imo.

Reply:
Done